erpnext/accounts/doctype/subscription/subscription.py

```diff
@@ -345,13 +345,14 @@ def create_invoice(self, prorate):
 			invoice.set_taxes()
 
 		# Due date
-		invoice.append(
-			'payment_schedule',
-			{
-				'due_date': add_days(invoice.posting_date, cint(self.days_until_due)),
-				'invoice_portion': 100
-			}
-		)
+		if not frappe.get_value(self.party_type, self.party, 'payment_terms'):
```

Comment:
```suggestion
		if self.days_until_due:
```
Default payment terms template can be there in Company master, Customer and Supplier Group, Customer and Supplier, so this check won't work. If days_until_due is mentioned then that should be honoured otherwise the default template should be applied.